linq4j/src/test/java/org/apache/calcite/linq4j/test/ExpressionTest.java

```diff
@@ -1357,6 +1357,25 @@ public void checkBlockBuilder(boolean optimizing, String expected) {
         Expressions.toString(Expressions.constant(12.34, BigDecimal.class)));
   }
 
+  @Test void testObjectConstantExpression() {
+    assertEquals("true",
+        Expressions.toString(Expressions.constant(true, Object.class)));
+    assertEquals("(byte)100",
+        Expressions.toString(Expressions.constant((byte) 100, Object.class)));
+    assertEquals("(char)100",
+        Expressions.toString(Expressions.constant((char) 100, Object.class)));
+    assertEquals("(short)100",
+        Expressions.toString(Expressions.constant((short) 100, Object.class)));
+    assertEquals("100",
+        Expressions.toString(Expressions.constant(100, Object.class)));
+    assertEquals("100L",
+        Expressions.toString(Expressions.constant(100L, Object.class)));
+    assertEquals("100.0F",
+        Expressions.toString(Expressions.constant(100F, Object.class)));
+    assertEquals("100.0D",
+        Expressions.toString(Expressions.constant(100D, Object.class)));
+  }
+
```

Comment:
The following two test case also pass without this commit. We can remove them.
```
    assertEquals("true",
        Expressions.toString(Expressions.constant(true, Object.class)));
    assertEquals("100",
        Expressions.toString(Expressions.constant(100, Object.class)));
```

Reply:
Done.